tools/alara_output_processing/alara_output_plotting.py

```diff
@@ -0,0 +1,285 @@
+import matplotlib.pyplot as plt
+
+#------------------------ Plotting Helper Functions --------------------------
+
+def preprocess_data(df_dicts, data_key='Data', seconds=True):
+    '''
+    Extract all of the ALARADFrames from df_dicts, their (shared) time
+        columns and store them in a list of tuples for each plot to be
+        produced. Additionally, for each plot, establish a corresponding
+        colormap.
+
+    Arguments:
+        df_dicts (dict or list): Single dictionary containing an ALARA output 
+            ALARADFrame and its metadata, of the form:
+            df_dict = {
+                'Run Label'   : (Distinguisher between runs),
+                'Variable'    : (Any ALARA output variable, dependent on ALARA
+                                 run parameters),
+                'Unit'        : (Respective unit matching the above variable),
+                'Data'        : (ALARADFrame containing ALARA output data for
+                                 the given run parameter, variable, and unit)
+            }
+        data_key (str, optional): Keyword for df_dicts to direct to the
+            desired ALARADFrame to source the plot's data.
+            (Defaults to 'Data')
+        seconds (bool, optional): Option to convert cooling times from
+            years to seconds.
+            (Defaults to True)
+
+    Returns:
+        all_data (list of tuples): A list of all relevant data for each table
```

Comment:
What if instead of returning a list of tuples, this method added a new entry for the `times` to the `df_dict` for each entry in `df_dicts`, and collected the labels to make the `color_map`.

Notwithstanding the discussion about putting some of the metadata into the tables, I like the idea of having the `df_dicts` cache the processed metadata rather than making new data structures that may be less transparent.